apps/passport-client/components/modals/AccountImportModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { Spacer } from "@pcd/passport-ui";
+import { PCD } from "@pcd/pcd-types";
+import { useCallback, useEffect, useState } from "react";
+import styled from "styled-components";
+import { useFilePicker } from "use-file-picker";
+import { useDispatch, usePCDCollection } from "../../src/appHooks";
+import { Button } from "../core";
+
+type ImportState =
+  | {
+      valid: true;
+      imported: false;
+      pcds: PCD[];
+      folders: FolderMap;
+      conflicts: number;
+    }
+  | { valid: true; imported: true; added: number }
+  | { valid: false };
+
+type FolderMap = { [pcdId: string]: string };
+
+export function AccountImportModal() {
+  const { openFilePicker, filesContent } = useFilePicker({
+    accept: ".json",
+    multiple: false
+  });
+
+  const pcdCollection = usePCDCollection();
+  const dispatch = useDispatch();
+
+  const [importState, setImportState] = useState<ImportState | undefined>();
+
+  const importPCDs = useCallback(() => {
+    // Should never happen, but makes TypeScript happy that we checked for it
+    if (!importState.valid || importState.imported === true) return;
+
+    const addedPCDs: PCD[] = [];
+    for (const pcd of importState.pcds) {
+      if (!pcdCollection.hasPCDWithId(pcd.id)) {
+        try {
+          pcdCollection.add(pcd);
+          if (importState.folders[pcd.id]) {
+            pcdCollection.setFolder(pcd.id, importState.folders[pcd.id]);
+          }
+
+          addedPCDs.push(pcd);
+        } catch (e) {
+          console.log(e);
+        }
+      }
+
+      setImportState({ valid: true, imported: true, added: addedPCDs.length });
+    }
+  }, [importState, pcdCollection]);
+
+  useEffect(() => {
+    (async () => {
+      if (filesContent.length > 0) {
```

Comment:
Someone is going to try importing someone else's export, so we should do some testing and think through whether we're okay with the results.  Cases I think might be important:

- Semaphore ID: You'll get someone else's private key.  We should test to make sure no logic elsewhere gets confused by this and uses the wrong one.  All the cases I know of (in UI and in login) are order-dependent, so I don't think they'll break, but I don't know the logic which looks up your ID for proving.
- Tickets: You'll have someone else's tickets, but they'll get deleted by reissuance if that's still turned on.  Making proofs about them may or may not work.  You probably also imported the semaphore ID which went with them, so depending how the lookup code works it might find the right one.
- Frogs: Same logic as tickets.  In this case, "cheating" seems most likely, and we should think about whether Frogcrypto's UI code should maybe display non-owned frogs differently.

I think "someone else's semaphore ID" is a potential big can of worms, so we should consider simply blocking the import entirely, or discarding the semaphore ID from the import if it doesn't match the user's identity.  Importantly, if we don't have one (because it was lost in a sync error) we should allow it to be imported to restore from backup.

Reply:
I've added a filtering option, which can be used to filter out certain PCDs during merging, and this is currently set to filter out semaphore identity PCDs, so they cannot be imported.

That said, if you have a copy of someone else's export, you could always directly insert it into your local storage, so all we can do is make it a bit harder to do things that don't make sense.

Reply:
A thought occurs to me (without having reviewed the latest code yet): This should probably be limited to filtering out an extra semaphore ID, but not disallowing importing a semaphore ID from backup if you somehow lost your primary one.  That would be important if we want import to be a way of recovering from full PCD loss.  Realistically, that recovery scenario might also need a separate interface to import at the login screen, since if the user doesn't happen to have a still-logged-in device they can't log in fully without their identity PCD.

Reply:
As of now, two rules are enforced: you can't import a semaphore identity PCD if you already have one, and you can't import a semaphore identity PCD that does not match your user account commitment, irrespective of whether you do or do not have a semaphore identity PCD already. There's also a catch-all rule that you can't import a PCD with an already-existing ID, which would prevent importing a semaphore identity PCD that you already have (with the caveat that I suppose one could manually edit the ID in the backup file, and so on).

I'm not intimately familiar with the classes of issues around lost semaphore identities, so I'm not sure if this is the right set of rules. Perhaps it should always be possible to import a semaphore identity PCD that matches your commitment on the user object, even if you already (for some mysterious reason) have some other semaphore identity PCD?